From d6096d863d9cfca763261968bdf817f516927078 Mon Sep 17 00:00:00 2001
From: Abhijith Das <adas@redhat.com>
Date: Fri, 15 Aug 2014 19:56:43 -0400
Subject: [fs] gfs2: Fix return value in slot_get()

Message-id: <1408132603-52384-18-git-send-email-adas@redhat.com>
Patchwork-id: 87820
O-Subject: [RHEL7.1 PATCH 17/17] GFS2: Fix return value in slot_get()
Bugzilla: 1026406
RH-Acked-by: Robert S Peterson <rpeterso@redhat.com>
RH-Acked-by: Steven Whitehouse <swhiteho@redhat.com>

ENOSPC was being returned in slot_get inspite of successful
execution of the function. This patch fixes this return
code.

Resolves: rhbz#1026406
Signed-off-by: Abhi Das <adas@redhat.com>
Signed-off-by: Steven Whitehouse <swhiteho@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/gfs2/quota.c b/fs/gfs2/quota.c
index aedd1d6..c9e262b 100644
--- a/fs/gfs2/quota.c
+++ b/fs/gfs2/quota.c
@@ -321,6 +321,7 @@ static int slot_get(struct gfs2_quota_data *qd)
  if (bit < sdp->sd_quota_slots) {
   set_bit(bit, sdp->sd_quota_bitmap);
   qd->qd_slot = bit;
+  error = 0;
 out:
   qd->qd_slot_count++;
  }
-- 
1.7.1